### ISSUE TITLE
Sync is stuck at 99%

### ISSUE THREAD

**Issue by arky** [digidem/mapeo-desktop#304]
Recurring sync issue with device sync stuck at 99%.  This issue is happening on multiple devices without any apparent error message or time outs. There weren't obvious error messages in the developer console, but I didn't dig too deeply.
 
![DSC_0148](https://user-images.githubusercontent.com/808895/76473707-f4b38b00-642b-11ea-9ddb-4da791759255.JPG)
![DSC_0147](https://user-images.githubusercontent.com/808895/76473744-0bf27880-642c-11ea-998c-17254eceac21.JPG)

I have shared a dataset for testing via email @gmaclennan  as well.

**Comment by noffle** [comment]
What versions of Mapeo Mobile are you using?

**Comment by arky** [comment]
Latest version of Mapeo Mobile is installed from Google Play

**Comment by arky** [comment]
@noffle Do you have any suggestions on how to collect specific errors messages from adb logs and developer console to aid debugging this problem. Thanks!

**Comment by noffle** [comment]
@arky If you feel comfortable with the command line, running Mapeo Desktop with the environment variable `DEBUG` set to `*` will cause all debug logs to be written to the terminal (on `standard error`). You can send it to a file with `2> debug.log`. I *think* the full command will look something like

```
set DEBUG=multifeed,blob-store-replication-stream,mapeo-core,hypercore-protocol & ./Mapeo.exe 2> debug.log
```

and then reproduce the sync hanging. 

@gmaclennan can you share the dataset with me? I can try and reproduce also.

**Comment by arky** [comment]
Thanks @noffle I'll get some debugging info this weekend. I'll also give you a datadump to debug. Meanwhile you could try to reproduce the problem with this datadump. 

https://gofile.io/?c=3GUV43

**Comment by arky** [comment]
@noffle Strange, now the sync is stuck doesn't even start. I am testing with all versions mapeo-deskto 5.x series on linux and windows now. 

https://paste.ubuntu.com/p/zDJkmZbw9V/

**Comment by arky** [comment]
<details>
<summary>
@noffle  Sync failed to start on Ubuntu with latest build of mapeo-desktop (v5.0.7)

https://paste.ubuntu.com/p/qVQt9sfNFq/
</summary>
![Screenshot from 2020-03-19 19-21-56](https://user-images.githubusercontent.com/808895/77067323-5faa2680-6a17-11ea-8210-1b9658343e49.png)
</details>

**Comment by noffle** [comment]
@arky gofile has been down the last couple days. Is there somewhere you can upload it?

**Comment by arky** [comment]
@noffle You should be able to get access to the files without a dropbox account or installing the dropbox app. 

https://www.dropbox.com/sh/d1l2i114f5932ja/AADux_oLM3m-Ou-JcL1UjkyHa?dl=0

**Comment by noffle** [comment]
@arky What are each of these files & how do they relate to the sync bug you're experiencing?

**Comment by arky** [comment]
@noffle If you load this dataset into your mapeo-desktop. And then try to sync it to mapeo-mobile you might be able to reproduce this bug.

**Comment by noffle** [comment]
I believe this is caused by [a bug in multifeed](https://github.com/kappa-db/multifeed/commit/0bc9bd99682c65343b23a0fc1ced2a358f7bee20) that was fixed on Jan 31. All versions of Mapeo Mobile that are 2.1.0-alpha.3 or older will be affected by this bug. The only release with the fix so far released is 2.1.0-alpha.4. For Mapeo Desktop, you'll need 5.0.1 or newer.

**Comment by okdistribute** [comment]
This should be fixed now, please re-open if you come across it!
